aws/resource_aws_appautoscaling_policy.go

```diff
@@ -285,10 +286,24 @@ func resourceAwsAppautoscalingPolicyCreate(d *schema.ResourceData, meta interfac
 }
 
 func resourceAwsAppautoscalingPolicyRead(d *schema.ResourceData, meta interface{}) error {
-	p, err := getAwsAppautoscalingPolicy(d, meta)
+	var p *applicationautoscaling.ScalingPolicy
+
+	err := resource.Retry(2*time.Minute, func() *resource.RetryError {
+		var err error
+		p, err = getAwsAppautoscalingPolicy(d, meta)
+		if err != nil {
+			if awsErr, ok := err.(awserr.Error); ok {
```

Comment:
Sorry I was not clear earlier. We can still use the `isAWSErr()` helper here to keep the code cleaner -- I'll fix these on merge.

```go
if isAWSErr(err, applicationautoscaling.ErrCodeFailedResourceAccessException, "") {
```